### DISCUSSION TITLE
TypeError when unmounting Vue component with an unused Teleport

### DISCUSSION THREAD

**Discussion by MetRonnie** [cypress-io/cypress#27180] [Component Testing]
### Problem

I have a component that includes
```vue
<Teleport to="#element-in-parent-component">
  ...
</Teleport>
```
 In the component test I am just mounting the component without the parent component; the teleport is unused. However this results in a TypeError on unmount causing the next test to fail
- Chrome: `TypeError: Cannot read properties of null (reading 'type')`
- Firefox: `TypeError: instance is null`

### Solution

Add a prop called something like `teleportTarget` which defaults to `null`, only passing in `#element-in-parent-component` from the parent component. And add a `v-if` condition
```vue
<Teleport v-if="teleportTarget" :to="teleportTarget">
  ...
</Teleport>
```
This way it doesn't bother trying to teleport during the component test

**Comment by jdbdnz** [comment]
Seems to be a Vue issue where components are destroyed before being passed to `unmountComponent`.

More context: https://github.com/vuejs/core/issues/9270
